python_modules/dagster/dagster/_core/execution/context/invocation.py

```diff
@@ -223,7 +256,13 @@ def set_requires_typed_event_stream(self, *, error_message: Optional[str]) -> No
         self._typed_event_stream_error_message = error_message
 
 
-class RunlessOpExecutionContext(OpExecutionContext):
+class RunlessRunProperties(RunProperties):
```

Comment:
I'm not feeling particularly confident in this implementation. 

For direct invocation, there is no `DagsterRun` and we need to throw an error if the user tries to access this object. For op context we handle this by throwing an error if a user accesses the `context.dagster_run` property. But since in the asset context, the `dagster_run` is fetched via the `run_properties` sub-object, we need a way to make that access throw an error. This was the best/only way i could find to do this after some tinkering, but i'm not convinced it's the best option

Reply:
no longer relevant, the implementation has been updated to not have a run properties sub object